### PR TITLE
Enable external ocp cluster-sync

### PR DESCRIPTION
**Pull request by qinqon**
**Is this a BUG FIX or a FEATURE ?**:

/kind enhancement
/kind infraestructure

**What this PR does / why we need it**:
Right now is not possible to deploy with `make cluster-sync` to an external system k8s or ocp, this PR use the KUBEVIRT_PROVIDER=external to do so and using the selected IMAGE_REPO/IMAGE_REGISTRY as intermediate repo.

**Special notes for your reviewer**:

You need to login at selected IMAGE_REGISTRY and IMAGE_REPO

**Release note**:

```release-note
Allow `make cluster-sync` on external systems.
```

### CODE REVIEW

File: cluster/sync.sh [hunk 1]
```diff
@@ -47,10 +47,19 @@ function isHandlerOk {
...
+    [[ "${KUBEVIRT_PROVIDER}" == external ]]
+}
 
+function isOpenshift {
+    $kubectl get co openshift-apiserver
+}
+
+function push() {
+    if isExternal; then
+        make manifests push
```
[the remark this answers is not in the document]
Comment (qinqon) [reply cluster/sync.sh:60]: like DEV_IMAGE_REGISTRY|DEV_IMAGE_REPO ?

[the remark this answers is not in the document]
Comment (phoracek) [reply cluster/sync.sh:60]: Sounds good

File: deploy/openshift/scc.yaml [hunk 2]
```diff
@@ -6,6 +6,10 @@ metadata:
 allowPrivilegedContainer: true
 allowHostDirVolumePlugin: true
 allowHostNetwork: true
+allowHostIPC: false
```
Comment (phoracek) [inline deploy/openshift/scc.yaml:9]: Unrelated

File: cluster/sync.sh [hunk 3]
```diff
@@ -47,10 +47,23 @@ function isHandlerOk {
...
-    make cluster-clean
+function isExternal {
+    [[ "${KUBEVIRT_PROVIDER}" == external ]]
+}
 
+function isOpenshift {
+    $kubectl get co openshift-apiserver
+}
+
+function push() {
```
Comment (phoracek) [inline cluster/sync.sh:58]: Please add howto use this in https://github.com/nmstate/kubernetes-nmstate/blob/master/CONTRIBUTING.md
